lib/cloud_controller/kpack/stager.rb

```diff
@@ -75,6 +75,11 @@ def image_resource(staging_details)
             blob: {
               url: blobstore_url_generator.package_download_url(staging_details.package),
             }
+          },
+          build: { 
+            env: staging_details.environment_variables.to_a.
+                  select{ | key, value | key =~ /^BP[A-Z]{0,1}_/ }.
```

Comment:
I'm curious why there is a whitelist on environment variables prefixed with `BP*`?

Reply:
This white list is not really required. It was only a gut feeling that feeding in variables like `VCAP_SERVICES` could cause problems. We also had these kind of problems when the `to_s` was missing in the line below.

Reply:
Yeah `VCAP_SERVICES` is not going to be the future method that bound services supply credentials to the app. There is upcoming work for this here - https://www.pivotaltracker.com/story/show/172562308

I think I'd still suggest relaxing the constraint, and maybe only filter out `VCAP_SERVICES`. There are many other environment variables that app developers set, and `BP*` is overly restrictive (and perhaps too specific to a single buildpack like bellsoft-liberica buildpack).